src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateCopy = { ...state };
+  const arrOfClones = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, { ...action.extraData });
```

Comment:
```suggestion
        Object.assign(stateCopy, action.extraData);
```
You can simplify in this case